csharp/src/Converters.cs

```diff
@@ -392,7 +392,11 @@ internal class XenDateTimeConverter : IsoDateTimeConverter
         public override object ReadJson(JsonReader reader, Type objectType, object existingValue, JsonSerializer serializer)
         {
             JToken jToken = JToken.Load(reader);
-            return DateTime.ParseExact(jToken.ToString(), DateFormats, CultureInfo.InvariantCulture, DateTimeStyles.None);
+            var str = jToken.ToObject<string>();
+
+            return str == null
```

Comment:
Shouldn't use `string.IsNullOrEmpty`?

Reply:
Hmm, yes, actually we should probably wrap the whole parsing in a try-catch block.